pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1260,10 +1260,13 @@ self: super: {
   ] super.svgcairo;
 
   # Espial is waiting for a hackage release to be compatible with GHC 9.X.
-  espial = appendPatch (fetchpatch {
-    url = "https://github.com/jonschoning/espial/commit/70177f9efb9666c3616e8a474681d3eb763d0e84.patch";
-    sha256 = "sha256-aJtwZGp9DUpACBV0WYRL7k32m6qWf5vq6eKBFq/G23s=";
-    excludes = ["package.yaml" "stack.yaml" "stack.yaml.lock"];
+  espial = overrideCabal (drv: {
+    src = pkgs.fetchFromGitHub {
+      owner = "jonschoning";
+      repo = "espial";
+      rev = "6654213418461df7addac8a1bc2d271c23cf82cb";
+      sha256 = "sha256-JXuk3d92/dnyL6d7LQDARD/U4nDw2vUCvpLgGrprB10=";
+    };
```

Comment:
We generally prefer applying patches, rather than overriding the source, since patches will (likely) break when a new version is released to Hackage.  When we override the source, we may forget about the override and end up many versions behind.

Would it be possible to get to the latest version with a patch file?  Also, could you create an issue on the repo asking for a Hackage release?

Reply:
Ah yeah, I was trying to apply an additional patch first to get things working and seeing this was behind thought I'd just replace the source, but what you said makes sense. I'll open an issue and update this to use another patch!

Reply:
@cdepillabout Actually, do you think it makes sense to just jailbreak this for now since the upper bound on `bytestring` is the only thing causing the build to fail? We can keep an eye on [the issue](https://github.com/jonschoning/espial/issues/49) and go with a local patch file if we don't hear back.

I could also just use this [patch](https://github.com/jonschoning/espial/compare/d56da21..6654213.patch) but then would have to have almost a dozen of files listed under excluded for it to work.

Reply:
In general: When a patch is not feasible a jailbreak + an assert on the current version is probably the right way to go.

Reply:
Alright, that's good know! Updated to assert the current version and jailbreak.